sympy/solvers/ode.py

```diff
@@ -277,10 +277,10 @@
 #: ``best``, and ``all_Integral`` meta-hints should not be included in this
 #: list, but ``_best`` and ``_Integral`` hints should be included.
 allhints = (
-    "separable",
-    "1st_exact",
     "1st_linear",
     "Bernoulli",
+    "separable",
+    "1st_exact",
```

Comment:
I examined the tests and it turns out that after reordering hints in dsolve, the code runs significantly faster in the test case:

>eq = (Eq(Derivative(x(t),t),x(t)*y(t)*sin(t)), Eq(Derivative(y(t),t),y(t)**2*sin(t)))
>dsolve(eq)

Without change:
1.53651309013 (time after execution - time at start of program)
real	0m1.818s 
user	0m1.774s
sys	0m0.044s

With change: (time after execution - time at start of program)
1.2372341156
real	0m1.567s
user	0m1.530s
sys	0m0.036s